### ISSUE TITLE
Add automatically generated API docs through Swagger

### ISSUE THREAD

**Issue by 0xThiebaut** [MISP/MISP#5848]
**Is your feature request related to a problem? Please describe.**
The [current MISP API documentation](https://www.circl.lu/doc/misp/automation/) is outdated... at best. Critical endpoints such as `/events/view/:id` or `/events/restSearch` are not even mentioned. The most stable approach to understand the MISP API is to actually [review the PyMISP source-code](https://github.com/MISP/PyMISP/blob/5cc7a1ad57c2e5a90092644e61c3de1e3e449a36/pymisp/api.py#L217) for hints such as:

```python
r = self._prepare_request('GET', f'events/view/{event_id}')
```

**Describe the solution you'd like**
I would like to suggest introducing automated [Swagger](https://swagger.io/) documentation which generates comprehensive API documentation as shown in their [demo Petstore API](https://petstore.swagger.io/).

![Screenshot_2020-05-02 Swagger UI](https://user-images.githubusercontent.com/46688461/80859030-f21c2780-8c5d-11ea-89b6-c91c33dbc760.png)

**Additional context**
I haven't investigated all CakePHP Swagger generators but these two seem interesting:
- [SwaggerBake plugin for CakePHP4](https://github.com/cnizzardini/cakephp-swagger-bake)
- [cakephp-swagger](https://github.com/alt3/cakephp-swagger)

Please, fix the current state of the API.

**Comment by RichieB2B** [comment]
That sounds like a great idea once we’re on CakePHP 4.x. Are we skipping 3.x?

**Comment by iglocska** [comment]
Indeed, the plan is to go straight for CakePHP 4.x. Timelines have been
shifted massively though due to all the lemons life has given us. Currently
busy making lemonade.

**Comment by iglocska** [comment]
With that said, /events/restSearch not being mentioned? You sure you're
looking in the right place? /events/automation is where the API
description. We have also started building our own self-documenting system
via the rest client built into MISP.

**Comment by 0xThiebaut** [comment]
I was unaware MISP's built-in REST client had that documentation. Thanks for pointing it out.

**Comment by packet-rat** [comment]
A general comment in support of the objective:  I've been using a number of APIs that provide comprehensive live documentation via Swagger.  I find this type of API Documentation extremely effective (especially the ones that provide **[Try it Out]**.    The current REST client endpoint (/servers/rest) is a GREAT Idea but  'quirky at best' and generally unusable.

**Comment by iglocska** [comment]
Unusable how? We use it for pretty much everything when it comes to hooking misp up to tools.

**Comment by SunnySkye** [comment]
Just to add to the numerous reasons why Swagger would be amazing for this... it actually makes integration with Azure LogicApps much better as well if the API has a Swagger endpoint.

LogicApps are obtuse at the best of times so anything to make it easier!

**Comment by 0xThiebaut** [comment]
Closing as stale, completed through OpenAPI documentation.
